bgpd/bgp_snmp.c

```diff
@@ -29,10 +29,125 @@
 #include "bgpd/bgp_snmp_bgp4.h"
 #include "bgpd/bgp_snmp_bgp4v2.h"
 #include "bgpd/bgp_mplsvpn_snmp.h"
+#include "bgpd/bgp_snmp_clippy.c"
+
+
+uint32_t bgp_snmp_traps_flags;
+
+static int bgp_cli_snmp_traps_config_write(struct vty *vty);
+
+
+DEFPY (bgp_snmp_traps_rfc4273,
+       bgp_snmp_traps_rfc4273_cmd,
+       "bgp snmp traps rfc4273<enable$on|disable$off>",
+       BGP_STR
+       "Configure BGP SNMP\n"
+       "Configure SNMP traps for BGP \n"
+       "Configure use of rfc4273 SNMP traps for BGP \n"
+       "Specify using of rfc4273 traps\n"
+       "Specify unusing rfc4273 traps\n") 
+{
+
+	if (on) {
+		SET_FLAG(bgp_snmp_traps_flags,
+				BGP_SNMP_TRAPS_RFC4273_ENABLED);
+		return CMD_SUCCESS;
+	}
+	if (off) {
+		UNSET_FLAG(bgp_snmp_traps_flags,
+			   BGP_SNMP_TRAPS_RFC4273_ENABLED);
+		return CMD_SUCCESS;
+	}
+
+	vty_out(vty, "bgp traps configuration must be specified \n");
+	return CMD_ERR_INCOMPLETE;
+}
+
+DEFPY (bgp_snmp_traps_bgp4_mibv2,
+       bgp_snmp_traps_bgp4_mibv2_cmd,
+       "bgp snmp traps bgp4-mibv2<enable$on|disable$off>",
+       BGP_STR
+       "Configure BGP SNMP\n"
+       "Configure SNMP traps for BGP \n"
+       "Configure use of bgp4-mibv2 SNMP traps for BGP \n"
+       "Specify using of bgp4-mibv2 traps\n"
+       "Specify unusing bgp4-mibv2 traps\n") 
+{
+
+	if (on) {
+		SET_FLAG(bgp_snmp_traps_flags,
+				BGP_SNMP_TRAPS_BGP4MIBV2_ENABLED);
+		return CMD_SUCCESS;
+	}
+	if (off) {
+		UNSET_FLAG(bgp_snmp_traps_flags,
+			   BGP_SNMP_TRAPS_BGP4MIBV2_ENABLED);
+		return CMD_SUCCESS;
+	}
+
+	vty_out(vty, "bgp traps configuration must be specified \n");
+	return CMD_ERR_INCOMPLETE;
+}
+
+static void bgp_snmp_traps_init(void)
+{
+	install_element(CONFIG_NODE, &bgp_snmp_traps_rfc4273_cmd);
+	install_element(CONFIG_NODE, &bgp_snmp_traps_bgp4_mibv2_cmd);
+
+	SET_FLAG(bgp_snmp_traps_flags,
+		 BGP_SNMP_TRAPS_RFC4273_ENABLED);
+	SET_FLAG(bgp_snmp_traps_flags,
+		 BGP_SNMP_TRAPS_BGP4MIBV2_ENABLED);	
+}
+
+int bgp_cli_snmp_traps_config_write(struct vty *vty)
+{
+	vty_out(vty, "!\n");
+	vty_out(vty, "traps rfc4273 %s\n",
+		CHECK_FLAG(bgp_snmp_traps_flags,
+			   BGP_SNMP_TRAPS_RFC4273_ENABLED) ?
+		"enabled":"disabled");
```

Comment:
Do we need to print these always? I mean regarding if it's disabled/enabled? Shouldn't we print only if it's a non-default?

Reply:
this will only occurs when snmp module is loaded

Reply:
I think that it only happening when the snmp module is loaded is kindof orthogonal, though ... when the configuration is generated, it should only create a line of config if the value is set to something other than the default (?)

Reply:
taken into account